vscode/syntaxes/c3.tmLanguage.json

```diff
@@ -37,23 +91,288 @@
 				{
 					"name": "keyword",
 					"match": "\\b(%switch|$default|$case|$if|$typeof|$else|$sizeof|$for|$case)\\b"
-				}
-			]
-		},
-		"comments": {
-			"patterns": [
+				},
+				{
+					"name": "keyword",
+					"match": "\\b(asm|assert|bitstruct|break|case|catch|const|continue)\\b"
+				},
+				{
+					"name": "keyword",
+					"match": "\\b(def|default|defer|distinct|do|else|enum|extern)\\b"
+				},
 				{
-					"name": "comment.line",
-					"begin": "//",
-					"end": "$"
+					"name": "keyword",
+					"match": "\\b(truefalse|fault|for|foreach|foreach_r|fn|tlocal|if)\\b"
```

Comment:
Should be `true|false`
Also true, false should be a constant like null, not a keyword.

Reply:
Fixed. And I eliminated some duplicated keywords.